src/components/useCurrenciesListHook.jsx

```diff
@@ -0,0 +1,29 @@
+import { useState, useEffect } from "react";
+import baseUrl from "../constants/baseUrl";
+
+function useCurrenciesList() {
+  const [list, setList] = useState([]);
+  const [error, setError] = useState(false);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    (async function () {
+      try {
+        setLoading(true);
+        fetch(` ${baseUrl}/currencies/eur.json`)
+          .then((res) => res.json())
+          .then((data) => {
+            setList(Object.keys(data.eur));
+          });
+      } catch (err) {
+        setError(err);
+      } finally {
+        setLoading(false);
+      }
+    })();
```

Comment:
Self invoking function is fine in this case, it would also be fine to name this function and call it right after, whatever you prefer :)

Reply:
VS Code does not have any objections when it has name in it, so I will use it :D